control-plane/api-gateway/controllers/utils.go

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllers
+
+import (
+	"strings"
+)
+
+func isModifiedError(err error) bool {
+	return strings.Contains(err.Error(), "the object has been modified")
```

Comment:
The pattern I typically see for this type of thing is to define the modified error as a variable and then to call Errors.Is. Something like the following.

```
var ErrModifiedError = errors.New("the object has been modified")
errors.Is(err, ErrDivideByZero)
```

But I'm not 100% on if the package throwing this error has a error defined already

Reply:
Yeah actually I think they do. Will update.

Reply:
Hmm actually it might just be a specific error message, https://github.com/kubernetes/kubernetes/blob/c95b16b28076e84867ae0521574303ffc058becf/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L240C2-L240C24

Reply:
https://github.com/kubernetes/apiserver/blob/v0.27.3/pkg/registry/generic/registry/store.go#L654

Reply:
It looks like it would just be a conflict error though, I don't think there is a specific one for this type. So I could just call 
```
if k8serrors.IsConflict(err) {
```
but that might catch other conflict error types.

Reply:
so I think we'll be good to just use `IsConflict` as you mention -- any time we have a conflict I'd imagine we'd want to retry.